### PR TITLE
Fix lumitech brightness and temperature

### PR DESCRIPTION
**Pull request by Arcticon**
Fixes UncleSamSwiss/ioBroker.loxone#150

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export type StateChangeListenEntry = {
     listener: StateChangeListener;
     opts?: StateChangeListenerOpts;
     queuedVal: ioBroker.StateValue | null;
-    ackTimer: ioBroker.Timeout | null;
+    ackTimer: ioBroker.Timeout | undefined;
```
Comment (raintonr) [inline src/main.ts:36]: Think I have mentioned before: please don't change null to undefined.

File: src/controls/ColorpickerBase.ts [hunk 2]
```diff
@@ -8,6 +8,13 @@ export abstract class ColorpickerBase extends ControlBase {
     private colorUpdateTimer?: NodeJS.Timeout;
 
     async loadColorPickerControlBaseAsync(uuid: string, control: Control): Promise<void> {
+        this.adapter.log.info(
```
Comment (raintonr) [inline src/controls/ColorpickerBase.ts:11]: This looks like debug, not info.

**Review by raintonr — CHANGES_REQUESTED** [review]

File: src/controls/ColorpickerBase.ts (hunk 2 above)
Comment (Arcticon) [reply src/controls/ColorpickerBase.ts]: i need to remove this

File: src/main.ts (hunk 1 above)
Comment (raintonr) [reply src/main.ts:36]: Nope. As discussed has to be undefined now.

**Review by raintonr — APPROVED** [review]

**Comment by raintonr** [comment]
I don't see why one needs to try `lumitechColorToBrightnessTemperature` and then fall back to `loxoneColorToBrightnessTemperature`.

Surely either the control is Lumitech or it is not? Is there some error or other logic in the Loxone Miniserver that sometimes reports the wrong control type? The referenced issue doesn't seem to mention this.

**IF** it turns out that the Loxone Miniserver sometimes incorrectly reports the control type then personally I would have merged `loxoneColorToBrightnessTemperature` & `lumitechColorToBrightnessTemperature` into one function and have that try to match `(lumitech|temp)\((\d+),(\d+)\)`.

**Comment by Arcticon** [comment]
you are correct. will change

**Comment by raintonr** [comment]
> the loxone UI always reported `temp(x,y)` instead of `lumitech(x,y)`. i sadly dont know why 🫤

Yeah. Loxone are annoying sometimes :disappointed: 

At least you have moved the 'or' logic from each call into `lumitechOrLoxoneColorToBrightnessTemperature` though so about the best we can do without Loxone actually doing what it says it will.

**Pull request merged**